src/test/validation_tests.cpp

```diff
@@ -57,15 +57,31 @@ BOOST_AUTO_TEST_CASE(subsidy_limit_test)
 {
     const auto chainParams = CreateChainParams(*m_node.args, ChainType::MAIN);
     CAmount nSum = 0;
-    for (int nHeight = 0; nHeight < 14000000; nHeight += 1000) {
+    auto step = 1000;
+    assert(chainParams->GetConsensus().nSubsidyHalvingInterval % step == 0);
+    for (int nHeight = 0; nHeight < 14000000; nHeight += step) {
         CAmount nSubsidy = GetBlockSubsidy(nHeight, chainParams->GetConsensus());
         BOOST_CHECK(nSubsidy <= 50 * COIN);
-        nSum += nSubsidy * 1000;
+        nSum += nSubsidy * step;
         BOOST_CHECK(MoneyRange(nSum));
     }
     BOOST_CHECK_EQUAL(nSum, CAmount{2099999997690000});
 }
 
+// Similar to subsidy_limit_test, but sums every single block subsidy until it drops to zero
+BOOST_AUTO_TEST_CASE(subsidy_sum_test)
+{
+    auto main_consensus{CreateChainParams(*m_node.args, ChainType::MAIN)->GetConsensus()};
+    CAmount total_subsidy_sats{0};
+    int block_height{0};
+    while (auto subsidy = GetBlockSubsidy(block_height, main_consensus)) {
```

Comment:
nit: Did you intentionally make it like this for clarity over
```suggestion
    while (auto subsidy{GetBlockSubsidy(block_height, main_consensus)}) {
```
?

Reply:
yeah, I'd use the `subsidy{}` in prod code, here I wanted the simpler syntax that laymen would also understand when looking at the test - I consider this the entrypoint for their don't-trust-verify journey.

Reply:
Would you consider this more readable?
```C++
for (CAmount subsidy; (subsidy = GetBlockSubsidy(block_height, main_consensus)) > 0; block_height++) {
    total_subsidy_sats += subsidy;
}
```

Reply:
Nah, to me that's more complicated. Better keep it as-is.